### PR TITLE
Fix Outstanding Bugs

### PR DESCRIPTION
**Pull request by Varhagna**
## Fixes
* Fix AirBlast, FireBlast, and AirSwipe going through walls
* Fix FireBurst not doing damage.
* Fix GeneralMethods#dropItems not checking for invalid drops.

## Misc. Changes
* Reduce collision radii for most abilities..

### CODE REVIEW

File: src/com/projectkorra/projectkorra/GeneralMethods.java [hunk 1]
```diff
@@ -650,7 +650,9 @@ public static void displayColoredParticle(final String hexVal, final Location lo
 	 */
 	public static void dropItems(final Block block, final Collection<ItemStack> items) {
 		for (final ItemStack item : items) {
-			block.getWorld().dropItem(block.getLocation(), item);
+			if(item.getType() != Material.AIR) {
+				block.getWorld().dropItem(block.getLocation(), item);continue;
```
Comment (Simplicitee) [inline src/com/projectkorra/projectkorra/GeneralMethods.java:654]: split this line into multiple lines pesant

File: src/com/projectkorra/projectkorra/GeneralMethods.java [hunk 2]
```diff
@@ -650,7 +650,9 @@ public static void displayColoredParticle(final String hexVal, final Location lo
 	 */
 	public static void dropItems(final Block block, final Collection<ItemStack> items) {
 		for (final ItemStack item : items) {
-			block.getWorld().dropItem(block.getLocation(), item);
+			if(item.getType() != Material.AIR) {
```
Comment (Simplicitee) [inline src/com/projectkorra/projectkorra/GeneralMethods.java:653]: should also check other materials that cannot be dropped

File: src/com/projectkorra/projectkorra/airbending/AirBlast.java [hunk 3]
```diff
@@ -206,12 +207,35 @@ private void advanceLocation() {
 		if (this.random.nextInt(4) == 0) {
 			playAirbendingSound(this.location);
 		}
-		if (GeneralMethods.checkDiagonalWall(this.location, this.direction)) {
+
+		BlockIterator blocks = new BlockIterator(this.getLocation().getWorld(), this.location.toVector(), this.direction, 0, (int) Math.ceil(this.direction.clone().multiply(speedFactor).length()));
+
+		while (blocks.hasNext() && checkLocation(blocks.next()));
```
Comment (Simplicitee) [inline src/com/projectkorra/projectkorra/airbending/AirBlast.java:213]: why is this while loop empty

File: src/com/projectkorra/projectkorra/GeneralMethods.java [hunk 4]
```diff
@@ -650,7 +650,10 @@ public static void displayColoredParticle(final String hexVal, final Location lo
 	 */
 	public static void dropItems(final Block block, final Collection<ItemStack> items) {
 		for (final ItemStack item : items) {
-			block.getWorld().dropItem(block.getLocation(), item);
+			if(item.getType() != Material.AIR) {
+				block.getWorld().dropItem(block.getLocation(), item);
+				continue;
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/GeneralMethods.java:655]: Is this here for a reason? It seems like you sorta did a short circuit and then changed it to a normal check.

File: src/com/projectkorra/projectkorra/firebending/FireBlast.java [hunk 5]
```diff
@@ -9,12 +9,14 @@
 import org.bukkit.block.BlastFurnace;
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.Levelled;
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/firebending/FireBlast.java:12]: Looks like this is unused.

File: src/com/projectkorra/projectkorra/firebending/FireBlast.java [hunk 6]
```diff
@@ -27,8 +29,10 @@
 import com.projectkorra.projectkorra.attribute.Attribute;
 import com.projectkorra.projectkorra.avatar.AvatarState;
 import com.projectkorra.projectkorra.command.Commands;
+import com.projectkorra.projectkorra.earthbending.lava.LavaFlow;
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/firebending/FireBlast.java:32]: Looks like this is unused.

File: src/com/projectkorra/projectkorra/firebending/FireBlast.java [hunk 7]
```diff
@@ -27,8 +29,10 @@
 import com.projectkorra.projectkorra.attribute.Attribute;
 import com.projectkorra.projectkorra.avatar.AvatarState;
 import com.projectkorra.projectkorra.command.Commands;
+import com.projectkorra.projectkorra.earthbending.lava.LavaFlow;
 import com.projectkorra.projectkorra.firebending.util.FireDamageTimer;
 import com.projectkorra.projectkorra.util.DamageHandler;
+import com.projectkorra.projectkorra.util.TempBlock;
```
Comment (ChristopherWMM) [inline src/com/projectkorra/projectkorra/firebending/FireBlast.java:35]: Looks like this is unused.

**Pull request merged**